crate_universe/src/config.rs

```diff
@@ -335,6 +335,87 @@ impl Sum for CrateAnnotations {
     }
 }
 
+/// A subset of `crate.annotation` that we allow packages to define in their
+/// free-form Cargo.toml metadata.
+///
+///     [package.metadata.bazel]
+///     additive_build_file_contents = """
+///         ...
+///     """
```

Comment:
I considered a more TOML-native form for `additive_build_file_content`, but just using multiline string literal containing Starlark seems all right.

Support for the structured form can be introduced separately if desired.

```toml
[package.metadata.bazel]
extra_aliased_targets = { cxx_cc = "cxx_cc" }
gen_build_script = false

[[package.metadata.bazel.additive_build_file_content]]
[package.metadata.bazel.additive_build_file_content.cc_library]
name = "cxx_cc"
srcs = ["src/cxx.cc"]
hdrs = ["include/cxx.h"]
include_prefix = "rust"
includes = ["include"]
linkstatic = true
strip_include_prefix = "include"
visibility = ["//visibility:public"]
```